clients/java/src/main/java/io/camunda/zeebe/client/impl/http/HttpClientFactory.java

```
@@ -110,8 +111,8 @@ private URI buildGatewayAddress() {
       builder.setScheme(config.isPlaintextConnectionEnabled() ? "http" : "https");
 
       return builder.build();
-    } catch (final URISyntaxException e) {
-      throw new RuntimeException(e);
+    } catch (final URISyntaxException | UnsupportedOperationException e) {
```

I need to replicate this in `io.camunda.client.impl.http.HttpClientFactory`
But I was wondering if we should rather add a fail fast in `CamundaClientBuilderImple#restAddress`, something like
```
 public CamundaClientBuilder restAddress(final URI restAddress) {
    if (restAddress != null && !restAddress.isAbsolute()) {
        // throw exception, illegal argument or similar
    }
    this.restAddress = restAddress;
    return this;
  }
```
and same for `grpcAddress` for the reasons explained in 
WDYT @koevskinikola @tmetzke ?

---

I think that makes sense. The URI won't work if it doesn't include the scheme, so we won't introduce a breaking change with this. We will just make it clearer to users what is going wrong.